astropy/io/ascii/ui.py

```diff
@@ -220,7 +220,7 @@ def _guess(table, read_kwargs, format, fast_reader):
             reader.guessing = True
             return reader.read(table)
 
-        except (core.InconsistentTableError, ValueError, TypeError,
+        except (core.InconsistentTableError, ValueError, TypeError, AttributeError,
```

Comment:
No, I wasn't implying anything of the like--I'm not expert enough on this code so I wasn't sure what existing exception class was most appropriate and so remained vague :)  But having hierarchies of more explicit exceptions doesn't hurt so long as they're sensible :)